google/resumable_media/requests/download.py

```diff
@@ -160,13 +171,48 @@ def consume(
         if self._stream is not None:
             request_kwargs["stream"] = True
 
+        # Assign object generation if generation is specified in the media url.
```

Comment:
Would this happen via a user specifying a generation on the object? Were we not respecting this previously?

Reply:
Yep this would happen via a user specifying a generation on the object. Previously, we've been respecting that only through `download.media_url`

A property `download._object_generation` is added. It records the object generation from either (1) generation query param from the media_url, or (2) the object generation from the initial response header. This specific line of code does (1) and retrieves it from the media_url

P.S. It's tricky in how limited information is passed from python-storage to resumable-media-python. A resumable-media-python `download` instance only knows the specified object generation from its `media_url`, and the "object" itself isn't pertained in a download.